lib/sensors/INA_226.cpp

```diff
@@ -0,0 +1,79 @@
+#include "mbed.h"
+#include "INA_226.h"
+
+INA_226::INA_226(int i2c_frequency, PinName SDA_pinname, PinName SCK_pinname, u_int8_t sensor_address, float max_expected_current, float shunt_resistance)
+{
+    shunt_resistance_ = shunt_resistance;
+    max_expected_current_ = max_expected_current;
+    sensor_address_ = sensor_address << 1;
+
+    i2c_(SDA_pinname, SCK_pinname);
+    i2c_.frequency(i2c_frequency);
+}
+
+INA_226::~INA_226(){}
+
+//TODO: may need to adjust function to take sensor slave addresses as there many be many current sensors
+float Safety_Board::getCurrentData()
+{
+    //may need char16_t?
+    u_int8_t cmd[3] = {CURRENT_REGISTER};
+    
+    i2c_.write(sensor_address_, cmd, 1);
+    i2c_.read(sensor_address_, cmd, 2, false);
+    
+    u_int16_t current_data = (cmd[2] << 8 | cmd[1]);
+

```

Comment:
bottom of pg 15 of data sheet, "The Current Register (04h) is then calculated by multiplying the decimal value of the Shunt Voltage Register (01h) contents by the decimal value of the Calibration Register and then dividing by 2048, as shown in Equation 3." And then this value should be multiplied to the current_LSB

Reply:
I'm a bit confused. Is it correct that the multiplication of the shunt voltage register by the Calibration register divided by 2048 is done and the value is but in the current register and what I need to do is multiply what is in the current register by the current_LSB?

Reply:
I think so